File: rehovot-ads-packages/ios/Plugin/InterstitialManager.swift
import Foundation

// Interstitial responsibilities are isolated so only this file changes for SDK swaps.
final class InterstitialManager {
    private let logger: Logger
    private var ready: Bool = false

    init(logger: Logger) {
        self.logger = logger
    }

    func load(_ options: InterstitialRequest) {
        logger.debug("TODO: Load Interstitial")
        logger.debug("Interstitial options: \(options)")
        ready = false
    }

    func show(_ options: InterstitialRequest) {
        logger.debug("TODO: Show Interstitial")
        logger.debug("Interstitial options: \(options)")
    }

    func isReady() -> Bool {
        return ready
    }

    func destroy() {
        logger.debug("TODO: Destroy Interstitial")
        ready = false
    }
}

File: rehovot-ads-packages/android/src/main/java/com/rehovottech/levelplay/InterstitialManager.kt
package com.rehovottech.levelplay

// Interstitial concerns stay isolated so the native SDK can be swapped later.
class InterstitialManager(
    private val logger: Logger,
) {
    private var ready: Boolean = false

    public fun load(options: InterstitialRequest) {
        logger.d("TODO: Load Interstitial")
        logger.d("Interstitial options: $options")
        ready = false
    }

    public fun show(options: InterstitialRequest) {
        logger.d("TODO: Show Interstitial")
        logger.d("Interstitial options: $options")
    }

    public fun isReady(): Boolean {
        return ready
    }

    public fun destroy() {
        logger.d("TODO: Destroy Interstitial")
        ready = false
    }
}
